src/read.c

```diff
@@ -4257,21 +4267,31 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
             maxItemID = item->id;
         }
         if (item->dimgForID == colorItem->id) {
+            avifBool seenAlphaForCurrentItem = AVIF_FALSE;
             for (uint32_t j = 0; j < meta->items.count; ++j) {
                 avifDecoderItem * auxlItem = &meta->items.item[j];
                 if (avifDecoderItemIsAlphaAux(auxlItem, item->id)) {
+                    if (seenAlphaForCurrentItem) {
+                        // Multiple items are claiming to be the alpha auxiliary of the current item.
+                        avifFree(alphaItemIndices);
+                        *alphaItem = NULL;
+                        *isAlphaItemInInput = AVIF_FALSE;
+                        return AVIF_RESULT_INVALID_IMAGE_GRID;
+                    }
                     alphaItemIndices[alphaItemCount++] = j;
+                    seenAlphaForCurrentItem = AVIF_TRUE;
```

Comment:
Should we also verify that `auxlItem->dimgForID == 0`?

Note that we will set `auxlItem->dimgForID` to the fake alpha grid item at line 4306 below. So we should make sure we don't overwrite an existing value. Also `auxlItem->dimgForID == 0` is implied by this diagram:

```
[primary item grid]
   ^       ^
   |dimg   |dimg
   |       |
[color] [color]
   ^       ^
   |auxl   |auxl
   |       |
[alpha] [alpha]
```

Reply:
Done.